src/Gui/TextEdit.cpp

```diff
@@ -359,12 +364,89 @@ void TextEditor::setSyntaxHighlighter(SyntaxHighlighter* sh)
     this->highlighter = sh;
 }
 
-void TextEditor::keyPressEvent (QKeyEvent * e)
+/** Sets the font, font size and tab size of the editor. */
+void TextEditor::OnChange(Base::Subject<const char*> &rCaller,const char* sReason)
+{
+    Q_UNUSED(rCaller);
+    ParameterGrp::handle hPrefGrp = getWindowParameter();
+    if (strcmp(sReason, "FontSize") == 0 || strcmp(sReason, "Font") == 0) {
+#ifdef FC_OS_LINUX
+        int fontSize = hPrefGrp->GetInt("FontSize", 15);
+#else
+        int fontSize = hPrefGrp->GetInt("FontSize", 10);
+#endif
+        QString fontFamily = QString::fromLatin1(hPrefGrp->GetASCII( "Font", "Courier" ).c_str());
+
+        QFont font(fontFamily, fontSize);
+        setFont(font);
+        lineNumberArea->setFont(font);
+    }
+    else {
+        QMap<QString, QColor>::Iterator it = d->colormap.find(QString::fromLatin1(sReason));
+        if (it != d->colormap.end()) {
+            QColor color = it.value();
+            unsigned int col = App::Color::asPackedRGB<QColor>(color);
+            auto value = static_cast<unsigned long>(col);
+            value = hPrefGrp->GetUnsigned(sReason, value);
+            col = static_cast<unsigned int>(value);
+            color.setRgb((col>>24)&0xff, (col>>16)&0xff, (col>>8)&0xff);
+            if (this->highlighter)
+                this->highlighter->setColor(QLatin1String(sReason), color);
+        }
+    }
+
+    if (strcmp(sReason, "TabSize") == 0 || strcmp(sReason, "FontSize") == 0) {
+        int tabWidth = hPrefGrp->GetInt("TabSize", 4);
+        QFontMetrics metric(font());
+        int fontSize = QtTools::horizontalAdvance(metric, QLatin1Char('0'));
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+        setTabStopWidth(tabWidth * fontSize);
+#else
+        setTabStopDistance(tabWidth * fontSize);
+#endif

```

Comment:
You can drop the check now that we no longer support Qt < 5.12.

Reply:
Okay, I will remove that.